nft/payment_obligation_integration_test.go

```diff
@@ -37,72 +49,169 @@ func TestMain(m *testing.M) {
+		Collaborators: []string{},
+		Data: &invoicepb.InvoiceData{
+			InvoiceNumber: "2132131",
+			GrossAmount:   123,
+			NetAmount:     123,
+			Currency:      "EUR",
+			DueDate:       &timestamp.Timestamp{Seconds: dueDate.Unix()},
+		},
+	})
+	assert.Nil(t, err, "should not error out when creating invoice model")
+	modelUpdated, txID, _, err := invSrv.Create(ctx, model)
+	err = txManager.WaitForTransaction(cid, txID)
+	assert.Nil(t, err)
+
+	// get ID
+	id, err := modelUpdated.ID()
+	assert.Nil(t, err, "should not error out when getting invoice ID")
+	// call mint
+	// assert no error
+	depositAddr := "0xf72855759a39fb75fc7341139f5d7a3974d4da08"
+	registry := cfg.GetContractAddress(config.PaymentObligation)
+
+	return ctx, id, registry, depositAddr, invSrv, cid
+}
+
+func mintNFT(t *testing.T, ctx context.Context, req nft.MintNFTRequest, cid identity.CentID, registry common.Address) nft.TokenID {
+	resp, done, err := payOb.MintNFT(ctx, req)
+	assert.Nil(t, err, "should not error out when minting an invoice")
+	assert.NotNil(t, resp.TokenID, "token id should be present")
+	tokenID, err := nft.TokenIDFromString(resp.TokenID)
+	assert.Nil(t, err, "should not error out when getting tokenID hex")
+	<-done
+	assert.NoError(t, txManager.WaitForTransaction(cid, uuid.Must(uuid.FromString(resp.TransactionID))))
+	owner, err := tokenRegistry.OwnerOf(registry, tokenID.BigInt().Bytes())
+	assert.NoError(t, err)
+	assert.Equal(t, req.DepositAddress, owner)
+	return tokenID
+}
+
+func TestPaymentObligationService_mint_grant_read_access(t *testing.T) {
+	t.SkipNow()
+	ctx, id, registry, depositAddr, invSrv, cid := prepareForNFTMinting(t)
+	req := nft.MintNFTRequest{
+		DocumentID:         id,
+		RegistryAddress:    registry,
+		DepositAddress:     common.HexToAddress(depositAddr),
+		ProofFields:        []string{"invoice.gross_amount", "invoice.currency", "invoice.due_date", "collaborators[0]"},
+		GrantNFTReadAccess: true,
+	}
+	tokenID := mintNFT(t, ctx, req, cid, registry)
+	doc, err := invSrv.GetCurrentVersion(ctx, id)
+	assert.NoError(t, err)
+	md, err := doc.PackCoreDocument()
+	assert.NoError(t, err)
+	cd := md.Document
+	assert.Len(t, cd.Roles, 2)
+	assert.Len(t, cd.Roles[1].Nfts, 1)
+	newNFT := cd.Roles[1].Nfts[0]
+	enft, err := documents.ConstructNFT(registry, tokenID.BigInt().Bytes())
+	assert.NoError(t, err)
+	assert.Equal(t, enft, newNFT)
+
+	// try to mint the NFT again
+	_, _, err = payOb.MintNFT(ctx, req)
+	assert.Error(t, err)
+	assert.True(t, errors.IsOfType(nft.ErrNFTMinted, err))
+}
+
+func failMintNFT(t *testing.T, grantNFT, nftReadAccess bool, roleProof []byte) {
+	ctx, id, registry, depositAddr, _, _ := prepareForNFTMinting(t)
+	req := nft.MintNFTRequest{
+		DocumentID:               id,
+		RegistryAddress:          registry,
+		DepositAddress:           common.HexToAddress(depositAddr),
+		ProofFields:              []string{"invoice.gross_amount", "invoice.currency", "invoice.due_date", "collaborators[0]"},
```

Comment:
Do we still need to use `collaborators[0]` here?

Reply:
removed it. Part of the test we are skipping